src/solver/optimisation/opt_gestion_des_couts_cas_lineaire.cpp

```diff
@@ -59,14 +60,14 @@ static void shortTermStorageCost(
             = CorrespondanceVarNativesVarOptim->SIM_ShortTermStorage.InjectionVariable[clusterGlobalIndex];
             varInjection >= 0)
         {
-            linearCost[varInjection] = 0;
+            linearCost[varInjection] = storage.spreadOnInjectionWithdrawal[hourInTheYear];
```

Comment:
About : 
`linearCost[varInjection] = storage.spreadOnInjectionWithdrawal[hourInTheYear];`

We know that we add a a spread on the injection here, so, to me, the following is enough : 
```C++
linearCost[varInjection] = storage.spread[hourInTheYear];
```
By the way, when you generated spreads, you called function **SIM_drawSpreadOnShortTermStorage**, without knowing if they apply to injection or withdrawal.

Reply:
We could even have : 
```C++
linearCost[varInjection] = storage.spread.generate();
```
**storage.spread** would be an object of class : 
```c++
const double EPSILON_ST_STORAGE = 1e+3;

class spread
{
public:
    spread() // Default constr
    {
        mt_.reset(seed_);
    }
    double generate() { return mt_.next() * EPSILON_ST_STORAGE;} 
private:
    int seed_ = ... ; // Whatever value
    MersenneTwister mt_;
}
```
The idea would be to have one spread object for all ST storage.

This would : 
- Avoid storing all spreads in a vector, and make RAM grow
- Simplify client code (we would not need **hourInTheYear** anymore, so an argument vanishes)
- We probably would need one seed only
- The spread class could maybe reused in another context
- In SIM, the call to **SIM_drawSpreadOnShortTermStorage** would disappear.